ops/pebble.py

```diff
@@ -702,12 +712,12 @@ class Layer:
     #: Mapping of check to :class:`Check` defined by this layer.
     checks: Dict[str, 'Check']
 
-    def __init__(self, raw: Optional[Union[str, 'LayerDict']] = None):
+    def __init__(self, raw: Optional[Union[str, '_LayerDict']] = None):
```

Comment:
Given the number of references to LayerDict, why are we making it 'private' ? I'm not sure if any of these should be prefixed with _ if the intent is that people are going to use them outside of that module.

Reply:
I actually really like this change. The Python ecosystem is always frantic, and `DeepDiff` requires a Rust compiler as of... sometime last week. "Is this dict equal to that dict?" gets really snarly when there are dicts nested under it, and encouraging to avoid it by making it more annoying is a plus from my point of view.

In addition, the fields under it are comprised of dunder types anyway.

Reply:
I wanted to make it private because I don't really want us to commit to the type annotations being stable just yet. But it's unnecessary churn and could potentially break people (even if only typing it doesn't seem worth it). So I've reversed that part of the change.